driver-examples/src/main/java/com/datastax/driver/examples/json/codecs/JacksonJsonCodec.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.driver.examples.json.codecs;
+
+import com.datastax.driver.examples.json.exceptions.InvalidTypeException;
+import com.datastax.oss.driver.api.core.ProtocolVersion;
+import com.datastax.oss.driver.api.core.type.DataType;
+import com.datastax.oss.driver.api.core.type.DataTypes;
+import com.datastax.oss.driver.api.core.type.codec.TypeCodec;
+import com.datastax.oss.driver.api.core.type.reflect.GenericType;
+import com.datastax.oss.driver.internal.core.util.Strings;
+import com.datastax.oss.protocol.internal.util.Bytes;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/**
+ * A JSON codec that uses the <a href="http://wiki.fasterxml.com/JacksonHome">Jackson</a> library to
+ * perform serialization and deserialization of JSON objects.
+ *
+ * <p>This codec maps a single Java object to a single JSON structure at a time; mapping of arrays
+ * or collections to root-level JSON arrays is not supported, but such a codec can be easily crafted
+ * after this one.
+ *
+ * <p>Note that this codec requires the presence of Jackson library at runtime. If you use Maven,
+ * this can be done by declaring the following dependency in your project:
+ *
+ * <p>
+ *
+ * <pre>{@code
+ * <dependency>
+ *   <groupId>com.fasterxml.jackson.core</groupId>
+ *   <artifactId>jackson-databind</artifactId>
+ *   <version>2.6.3</version>
```

Comment:
This is a rather old version, please use the latest one available.

Reply:
I've changed to use placeholder: `${jackson.version}`